scene/main/window.cpp

```diff
@@ -918,7 +935,7 @@ void Window::set_transient(bool p_transient) {
 		return;
 	}
 
-	if (transient) {
+	if (transient && !transient_to_focused) {
 		_make_transient();
 	} else {
 		_clear_transient();
```

Comment:
This will call `_clear_transient()` when you have `transient_to_focused` set to `true` and then set `transient` to `true`, I think.

Reply:
Indeed, it should call `_clear_transient` only if `transient` is unset, so should be:
```c++
if (transient) {
	if (!transient_to_focused) {
		_make_transient();
	}
} else {
	_clear_transient();
}
```

Reply:
Great catch, thanks!